x-pack/plugins/gis/public/meta.js

```diff
@@ -11,21 +11,29 @@ import _ from 'lodash';
 const GIS_API_RELATIVE = `../${GIS_API_PATH}`;
 
 let meta = null;
+let loadingMetaPromise = null;
 let isLoaded = false;
-export async function getDataSources() {
-  if (!meta) {
-    meta = new Promise(async (resolve, reject) => {
-      try {
-        const meta = await fetch(`${GIS_API_RELATIVE}/meta`);
-        const metaJson = await meta.json();
-        isLoaded = true;
-        resolve(metaJson.data_sources);
-      } catch(e) {
-        reject(e);
-      }
-    });
+export function getDataSources() {
```

Comment:
It can but that does not provide any value since `await` is never used inside the function block. It is still an asynchronous function since it can return a promise.